projects/core/src/cart/facade/cart-util.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Cart } from '../../model/cart.model';
+
+export class CartUtilService {
+  constructor() {}
+
+  static isCreated(cart: Cart): boolean {
+    return (
+      cart &&
+      !!Object.keys(cart).length &&
+      (typeof cart.guid !== 'undefined' || typeof cart.code !== 'undefined')
+    );
+  }
+
+  static isLoaded(cart: Cart): boolean {
+    // totalItems are checked to be sure we have loaded cart (if they are undefined, we loaded guid and code from localStorage, but didn't fetched cart from backend)
+    return (
+      cart && Object.keys(cart).length && typeof cart.totalItems !== 'undefined'
```

Comment:
This function is very confusing. After investigating, I think, it checks whether cart data is details or just basic. Checking "cart.totalItems" is not a good. To make things simple, we can remove "details" flag from `occ-cart.adapter`. So, cart details data is always loaded. So, there is no need check "details" any more.

Reply:
As discussed offline, it's to determine if cart is deficient due to being initialized parially by the local storage. The name of this function will be improved.